#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.23624887082405407 -0.25731267314299494 0.9370040871172151 4.703178361918595
  position -0.1539983359533472 1.0967505598498564 0.6370930656503582
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  translation 0 0 -0.1
  size 1 1
}
Solid {
  translation 0 -0.5 -0.05
  rotation 0 1 0 4.692820414042842e-06
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "red_brick_wall.jpg"
          ]
        }
      }
      geometry Box {
        size 1 0.01 0.1
      }
    }
  ]
  name "Wall1"
  boundingObject USE WALL_SHAPE
}
Solid {
  translation -0.5 0 -0.05
  rotation 0 0 1 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "Wall2"
  boundingObject USE WALL_SHAPE
}
Solid {
  translation 0 0.5 -0.05
  rotation 0 0 1 -3.1415853071795863
  children [
    USE WALL_SHAPE
  ]
  name "Wall3"
  boundingObject USE WALL_SHAPE
}
Solid {
  translation 0.5 0 -0.05
  rotation 0 0 1 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "Wall4"
  boundingObject USE WALL_SHAPE
}
Robot {
  translation 0 0 -0.06
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
          acceleration 1
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.05 0.06 0
        rotation 1 0 0 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
          acceleration 1
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation 0.05 -0.06 0
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid2"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
          acceleration 1
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation -0.05 0.06 0
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid3"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
          acceleration 1
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.05 -0.06 0
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid4"
        boundingObject USE WHEEL
      }
    }
  ]
  boundingObject USE BODY
  controller "obstacle_avoidance"
}
